api/src/main/java/org/openmrs/api/context/OpenmrsApplicationContextConfig.java

```diff
@@ -0,0 +1,217 @@
+            HL7Service hL7Service, MessageSourceService messageSourceService, VisitService visitService,
+            ProviderService providerService) {
+        ServiceContext serviceContext = ServiceContext.getInstance();
+        serviceContext.setPatientService(patientService);
+        serviceContext.setPersonService(personService);
+        serviceContext.setConceptService(conceptService);
+        serviceContext.setUserService(userService);
+        serviceContext.setObsService(obsService);
+        serviceContext.setEncounterService(encounterService);
+        serviceContext.setLocationService(locationService);
+        serviceContext.setOrderService(orderService);
+        serviceContext.setConditionService(conditionService);
+        serviceContext.setDiagnosisService(diagnosisService);
+        serviceContext.setMedicationDispenseService(medicationDispenseService);
+        serviceContext.setOrderSetService(orderSetService);
+        serviceContext.setFormService(formService);
+        serviceContext.setAdministrationService(adminService);
+        serviceContext.setDatatypeService(datatypeService);
+        serviceContext.setProgramWorkflowService(programWorkflowService);
+        serviceContext.setCohortService(cohortService);
+        serviceContext.setMessageService(messageService);
+        serviceContext.setSerializationService(serializationService);
+        serviceContext.setSchedulerService(schedulerService);
+        serviceContext.setAlertService(alertService);
+        serviceContext.setHl7Service(hL7Service);
+        serviceContext.setMessageSourceService(messageSourceService);
+        serviceContext.setVisitService(visitService);
+        serviceContext.setProviderService(providerService);
+        return serviceContext;
+    }
+
+    @Bean
+    public Context context(ContextDAO contextDAO, ServiceContext serviceContext) {
+        Context context = new Context();
+        context.setServiceContext(serviceContext);
+        context.setContextDAO(contextDAO);
+        return context;
+    }
+
+    @Bean
+    public List<Object> moduleTestingMappingJarLocations() {
+        return new ArrayList<>();
+    }
+
+    @Bean
+    public List<Object> mappingJarResources() {
+        List<Object> mergedList = new ArrayList<>(moduleTestingMappingJarLocations());
+        return mergedList;
+    }
+
+    @Bean
+    public AuditableInterceptor auditableInterceptor() {
+        return new AuditableInterceptor();
+    }
+
+    @Bean
+    public HibernateSessionFactoryBean sessionFactory(Resource mappingJarResources) {
+        HibernateSessionFactoryBean sessionFactory = new HibernateSessionFactoryBean();
+        sessionFactory.setConfigLocations(new Resource[] {
+                new ClassPathResource("hibernate.cfg.xml")
+        });
+        sessionFactory.setMappingJarLocations(mappingJarResources);
+        sessionFactory.setPackagesToScan("org.openmrs");
+        return sessionFactory;
```

Comment:
HibernateSessionFactoryBean is a strange mix of annotations and bean config, but we can refactor later on to annotations. I'm fine with this for now.